src/lib/common/sol-bus.c

```diff
@@ -383,9 +383,8 @@ sol_bus_client_free(struct sol_bus_client *client)
     if (!client)
         return;
 
-    destroy_client(client);
-
     sol_ptr_vector_remove(&_ctx.clients, client);
+    destroy_client(client);
```

Comment:
well, for that at least there is a reason: if you `a - b` and they are not in the same segment, then the result will not be stable.